crates/services/block_aggregator_api/src/blocks/importer_and_db_source.rs

```diff
@@ -1,143 +1,190 @@
 use crate::{
-    blocks::{
-        BlockSource,
-        BlockSourceEvent,
-        importer_and_db_source::importer_service::ImporterTask,
-    },
+    blocks::BlockSource,
     result::{
         Error,
         Result,
     },
 };
-use anyhow::anyhow;
-use fuel_core_services::{
-    Service,
-    ServiceRunner,
-    stream::BoxStream,
-};
 use fuel_core_storage::{
     Error as StorageError,
     StorageInspect,
-    tables::FuelBlocks,
+    tables::{
+        FuelBlocks,
+        Transactions,
+    },
 };
 use fuel_core_types::{
     blockchain::block::Block as FuelBlock,
+    fuel_tx::{
+        Receipt as FuelReceipt,
+        Receipt,
+        Transaction,
+        TxId,
+    },
     fuel_types::BlockHeight,
-    services::block_importer::SharedImportResult,
 };
-
-use crate::blocks::importer_and_db_source::sync_service::{
-    SyncTask,
-    TxReceipts,
-};
-use fuel_core_storage::tables::Transactions;
-use fuel_core_types::fuel_tx::Receipt as FuelReceipt;
-
-pub mod importer_service;
-pub mod sync_service;
-#[cfg(test)]
-mod tests;
+use std::sync::Arc;
 
 pub mod serializer_adapter;
 
-pub trait BlockSerializer {
+pub trait BlockSerializer: Send + Sync + 'static {
     type Block;
+
     fn serialize_block(
         &self,
         block: &FuelBlock,
-        receipts: &[FuelReceipt],
+        receipts: &[Vec<FuelReceipt>],
     ) -> Result<Self::Block>;
 }
 
-/// A block source that combines an importer and a database sync task.
-/// Old blocks will be synced from a target database and new blocks will be received from
-/// the importer
-pub struct ImporterAndDbSource<Serializer, DB, Receipts>
+pub trait TxReceipts: Send + Sync + 'static {
+    fn get_receipts(&self, tx_id: &TxId) -> Result<Vec<Receipt>>;
+}
+
+pub struct OldBlocksSource<Serializer, DB, Receipts> {
+    serializer: Arc<Serializer>,
+    db: Arc<DB>,
+    receipts: Arc<Receipts>,
+}
+
+impl<Serializer, DB, Receipts> OldBlocksSource<Serializer, DB, Receipts> {
+    pub fn new(serializer: Arc<Serializer>, db: DB, receipts: Receipts) -> Self {
+        Self {
+            serializer,
+            db: Arc::new(db),
+            receipts: Arc::new(receipts),
+        }
+    }
+}
+
+impl<Serializer, DB, Receipts> OldBlocksSource<Serializer, DB, Receipts>
 where
-    Serializer: BlockSerializer + Send + Sync + 'static,
-    <Serializer as BlockSerializer>::Block: Send + Sync + 'static,
     DB: Send + Sync + 'static,
     DB: StorageInspect<FuelBlocks, Error = StorageError>,
     DB: StorageInspect<Transactions, Error = StorageError>,
     Receipts: TxReceipts,
+    Serializer: BlockSerializer,
```

Comment:
Note: I think we should rename `BlockSerializer` before we merge all of this (unless we go back to storing blocks as bytes). It's not a serializer, more a converter haha.